pkg/catalog/catalog.go

```diff
@@ -1940,29 +1969,116 @@ func (c *Catalog) VerifyLinkAddress(ctx context.Context, repository, token strin
 }
 
 func (c *Catalog) DeleteExpiredLinkAddresses(ctx context.Context) {
-	it, err := c.Store.ListRepositories(ctx)
+	repos, err := c.listRepositoriesHelper(ctx)
 	if err != nil {
-		c.log.WithError(err).Warn("Failed to list repositories during delete expired addresses")
+		c.log.WithError(err).Warn("Failed list repositories during delete expired addresses")
 		return
 	}
+
+	for _, repo := range repos {
+		err := c.Store.DeleteExpiredLinkAddresses(ctx, repo)
+		if err != nil {
+			c.log.WithError(err).WithField("repository", repo.RepositoryID).Warn("Delete expired address tokens failed")
+		}
+	}
+}
+
+func (c *Catalog) listRepositoriesHelper(ctx context.Context) ([]*graveler.RepositoryRecord, error) {
+	it, err := c.Store.ListRepositories(ctx)
+	if err != nil {
+		return nil, err
+	}
 	defer it.Close()
 
 	var repos []*graveler.RepositoryRecord
 	for it.Next() {
-		record := it.Value()
-		repos = append(repos, record)
+		repos = append(repos, it.Value())
 	}
 	if err := it.Err(); err != nil {
-		c.log.WithError(err).Warn("Failed to iterate over repositories during delete expired addresses")
+		return nil, err
+	}
+	return repos, nil
+}
+
+func (c *Catalog) DeleteTrackedPhysicalAddresses(ctx context.Context) {
+	repos, err := c.listRepositoriesHelper(ctx)
+	if err != nil {
+		c.log.WithError(err).Warn("Failed list repositories during delete tracked physical addresses")
 		return
 	}
 
 	for _, repo := range repos {
-		err := c.Store.DeleteExpiredLinkAddresses(ctx, repo)
+		if err := c.deleteRepoTrackedPhysicalAddresses(ctx, repo); err != nil {
+			c.log.WithError(err).
+				WithField("repository", repo.RepositoryID).
+				Warn("Repository delete tracked physical addresses failed")
+		}
+	}
+}
+
+func (c *Catalog) deleteRepoTrackedPhysicalAddresses(ctx context.Context, repo *graveler.RepositoryRecord) error {
+	// collect and delete old keys - work in 'kvTrackDeleteBatchSize' and interval of 5 min between if batch is bigger
```

Comment:
You can skip the optimizations for kv calls at the moment, we need to handle this as part of the bigger picture. This is not the only place where this pose a problem.

Reply:
Suggest to remove the sleep or make the delete unbound to any size?